{
  String text=t.getMessage();
  Throwable cause=t;
  for (int i=0; i < MAX_EXCEPTION_DEPTH; ++i) {
    if (text != null || cause == null) {
      break;
    }
    text=cause.getMessage();
    cause=cause.getCause();
  }
  if (text == null) {
    text=String.format("An exception was encountered: %s (no message)",t.getClass().getName());
  }
 else {
    text=String.format("(%s): %s",t.getClass().getName(),text);
  }
  errorReported();
  addMessage(new Message(MessageKind.ERROR,text).setException(t));
}
